src/main/java/com/ncc/aif/ThreadedValidationEngine.java

```diff
@@ -0,0 +1,245 @@
+
+        @Override
+        public String toString() {
+            return String.join(" ", threadName, shapeName, duration + "ms",
+                    filteredTargetCount + "/" + targetCount, "v" + violations, ignored ? "ignored" : "");
+        }
+    }
+
+    @Override
+    public Resource createResult(Resource type, Constraint constraint, RDFNode focusNode) {
+        Resource result = threadModel.get().createResource(type);
+        result.addProperty(SH.resultSeverity, constraint.getShapeResource().getSeverity());
+        result.addProperty(SH.sourceConstraintComponent, constraint.getComponent());
+        result.addProperty(SH.sourceShape, constraint.getShapeResource());
+        if (focusNode != null) {
+            result.addProperty(SH.focusNode, focusNode);
+        }
+
+        // check whether this thread has enough violations to trigger exception
+        if (constraint.getShapeResource().getSeverity() == SH.Violation) {
+            int violations = threadViolations.get() + 1;
+            threadViolations.set(violations);
+            if (exceedsMaximumNumberViolations(violations)) {
+                throw new MaximumNumberViolations(violations);
+            }
+        }
+
+        return result;
+    }
+
+    @Override
+    public void setFocusNodeFilter(Predicate<RDFNode> value) {
+        super.setFocusNodeFilter(value);
+        focusNodeFilter = value;
+    }
+
+    private boolean exceedsMaximumNumberViolations(int violations) {
+        int errorBatch = getConfiguration().getValidationErrorBatch();
+        return errorBatch != -1 && violations >= errorBatch;
+    }
+
+    /**
+     * Validates all target nodes against all of their shapes. The provided {@code executor} is used to provide
+     * processing for each shape. This allows the user some control over the environment in which each shape is processed.
+     *
+     * To further narrow down which nodes to validate, use {@link #setFocusNodeFilter(Predicate)}.
+     *
+     * @param executor {@link ExecutorService} to send jobs to
+     * @return an instance of sh:ValidationReport in the results Model
+     * @throws InterruptedException when {@link Future#get()} experiences {@link InterruptedException}
+     * @throws ExecutionException when {@link Future#get()} experiences {@link ExecutionException}
+     */
+    public Resource validateAll(ExecutorService executor) throws InterruptedException, ExecutionException {
+        List<Shape> rootShapes = shapesGraph.getRootShapes();
+        //TODO: Add monitor support for threaded validator. Experience NPE with AIFProgressMonitor
+//        if (monitor != null) {
+//            monitor.beginTask("Validating " + rootShapes.size() + " shapes", rootShapes.size());
+//        }
+
+        int i = 0;
+        for (Shape shape : rootShapes) {
+            validationMetadata.add(executor.submit(getTask(shape, i++, executor)));
+        }
+
+        // Go through all futures and get md for those that have completed
```

Comment:
change `md` in comment to `validation metadata`

Reply:
fixed